relm4/src/component/builder.rs

```diff
@@ -66,15 +66,25 @@ where
 
 impl<C: Component> ComponentBuilder<C>
 where
-    C::Root: AsRef<gtk::Window>,
+    C::Root: AsRef<gtk::Window> + Clone,
 {
     /// Set the component's root widget transient for a given window.
     ///
     /// If the root widget is a native dialog, such as [`gtk::FileChooserNative`],
     /// you should use [`transient_for_native`][ComponentBuilder::transient_for_native] instead.
     #[must_use]
-    pub fn transient_for(self, window: impl AsRef<gtk::Window>) -> Self {
-        self.root.as_ref().set_transient_for(Some(window.as_ref()));
+    pub fn transient_for(self, widget: impl AsRef<gtk::Widget>) -> Self {
+        let root = widget.as_ref().root();
+        if let Some(root) = root {
```

Comment:
Also, what if this `root` isn't a window? In the [example](https://github.com/Relm4/Relm4/discussions/245#discussioncomment-3396902) it's actually a button.